### ISSUE TITLE
Notebook: Hyperparameter / Grid Search

### ISSUE THREAD

**Issue by DiogenesAnalytics** [DiogenesAnalytics/autoencoder#36]
# Problem
Would be worthwhile to have a notebook that demonstrates how to find the optimal parameters/architecture for a an autoencoder mode.

## References
+ https://github.com/CodeReclaimers/neat-python
+ https://keras.io/keras_tuner/
+ https://keras.io/guides/keras_tuner/
+ http://www.cs.ucf.edu/~kstanley/#publications
+ [Keras Tuner: Layer Type](https://chat.openai.com/share/a515e092-e173-4ac7-bebd-4638b471611f)
+ https://www.tensorflow.org/tutorials/keras/keras_tuner
+ [Minimize Max Recon Error](https://chat.openai.com/share/6ed4a754-9deb-4ebd-8c4c-cf1eb1e34afa)

**Comment by DiogenesAnalytics** [comment]
# MinNDAE Tuning
See: https://github.com/DiogenesAnalytics/autoencoder/blob/master/notebooks/tuning/minndae.ipynb

# Loss Function
```math
loss + encode * reg
```

# Plot
![minndae_optimal_encode_dim_784-1_reg](https://github.com/DiogenesAnalytics/autoencoder/assets/118541082/3238a43b-a9f8-4520-9094-375b283a5442)

**Comment by DiogenesAnalytics** [comment]
# MinNDAE Tuning
See: https://github.com/DiogenesAnalytics/autoencoder/blob/master/notebooks/tuning/minndae.ipynb

# Loss Function
```math
loss + (loss * encode * reg)
```

# Plot

![minndae_optimal_encode_dim_784-1_reg_default_penalty](https://github.com/DiogenesAnalytics/autoencoder/assets/118541082/d220035a-93d3-4654-a890-ab443900cf15)

**Comment by DiogenesAnalytics** [comment]
![integral_balance_point_minndae_optimal_code_dim](https://github.com/DiogenesAnalytics/autoencoder/assets/118541082/906ddab6-dcab-43be-ba83-a9e04c5e8e27)
![-xy_minndae_optimal_code_dim](https://github.com/DiogenesAnalytics/autoencoder/assets/118541082/54110c15-7dfd-4574-bfe4-1be91e5515d6)
